app/controllers/EmailByEoriController.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import scala.concurrent.ExecutionContext
+
+import connectors.CustomsDataStoreConnector
+import controllers.actions.AuthAction
+import javax.inject.{Inject, Singleton}
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+
+@Singleton
+class EmailByEoriController @Inject()(authorise: AuthAction, customsDataStoreConnector: CustomsDataStoreConnector, cc: ControllerComponents)(
+  implicit ec: ExecutionContext
+) extends BackendController(cc) {
+
+  def getEmailIfVerified(eori: String): Action[AnyContent] = authorise.async { implicit request =>
+    customsDataStoreConnector
+      .getEmailAddress(eori)
+      .map {
+        case Some(emailAddress) => Ok(Json.toJson(emailAddress))
+        case None               => NotFound
+      }
+      .recover { case _ => InternalServerError }
```

Comment:
I am wondering: is there a difference between using this `recover` method and having all exceptions being handled by the global handler? If I remember corrctly, it also returns 5xx status in case of exception.

Reply:
Not really. If you check the default ErrorHandler implementation (`uk.gov.hmrc.play.bootstrap.backend.http.JsonErrorHandler`) there's a specific `case` for `UpstreamErrorResponse` which results to `UpstreamErrorResponse.reportAs`.
This means that you could have for instance a `BAD_GATEWAY` status as a result and not always `INTERNAL_SERVER_ERROR`